src/engraving/libmscore/slur.cpp

```diff
@@ -1091,6 +1127,15 @@ SpannerSegment* Slur::layoutSystem(System* system)
             Chord* c1 = startCR()->isChord() ? toChord(startCR()) : 0;
             Chord* c2 = endCR()->isChord() ? toChord(endCR()) : 0;
 
+            if (_sourceStemArrangement != -1) {
+                if (_sourceStemArrangement != calcStemArrangement(c1, c2)) {
+                    // copy & paste from incompatible stem arrangement, so reset bezier points
+                    for (int g = 0; g < (int)Ms::Grip::GRIPS; ++g) {
+                        slurSegment->ups((Ms::Grip)g) = UP();
+                    }
```

Comment:
BTW currently this is only done for slur direction=AUTO (vs UP/DOWN), but might make sense to do in all cases.